sql-plugin/src/main/scala/com/nvidia/spark/rapids/RmmRapidsRetryIterator.scala

```diff
@@ -620,4 +638,15 @@ trait CheckpointRestore {
    * Restore state that was checkpointed.
    */
   def restore(): Unit
-}
\ No newline at end of file
+}
+
+/**
+ * This is a wrapper that turns a target size into an autocloseable to allow it to be used
+ * in withRetry blocks.  It is intended to be used to help with cases where the split calculation
+ * happens inside the retry block, and depends on the target size.  On a SplitAndRetryOOM,
+ * a split policy like `splitTargetSizeInHalf` can be used to retry the block with a smaller target
+ * size.
+ */
+case class AutoCloseableTargetSize(targetSize: Long, minSize: Long) extends AutoCloseable {
+  override def close() = {}
```

Comment:
nit: public methods are typically still annotated with types, and an explict `Unit` type constant can be used instead of an empty block

```suggestion
  override def close(): Unit = ()
```

Reply:
Will fix.